### PR TITLE
Add `has_generator` property to operation.

### PR DESCRIPTION
**Pull request by dwierichs**
**Context:**
Currently, we have the function `qml.operation.has_gen` that checks via a try-except clause whether the generator of an operation may be constructed. This can be very slow, given that we only want to know whether there is a generator implemented, which even is static class information (except for a few symbolic ops).

**Description of the Change:**
This PR adds `Operation.has_generator` in order to make a faster check available to `qml.operation.has_gen`.
It follows the implementation of the other `has_..` attributes, like `has_matrix` or `has_decomposition`.

**Benefits:**
Faster checks via `qml.operation.has_gen`

**Possible Drawbacks:**
Code to be maintained. This should be a minor drawback, given that there are only a few operations that have explicit `has_generator` additions.

**Related GitHub Issues:**
This will make #3874 feasible.

### CODE REVIEW

File: tests/test_operation.py [hunk 1]
```diff
@@ -451,6 +451,35 @@ class MyOp(qml.operation.Operator):
...
+    def test_has_generator_false(self):
+        """Test `has_generator` property defaults to false if `generator` not overwritten."""
+
+        class MyOp(qml.operation.Operator):
+            num_wires = 1
+
+        assert MyOp.has_generator is False
+        assert MyOp(wires=0).has_generator is False
+
+    def test_has_generator_false_concrete_template(self):
```
Comment (lillian542) [inline tests/test_operation.py:476]: Perhaps a bit redundant but maybe an analogous test for a concrete operation that _does_ have a generator defined could be added? E.g. RZ

**Review by lillian542 — APPROVED** [review]
Looks good to me! 🎉

File: tests/test_operation.py (hunk 1 above)
Comment (dwierichs) [reply tests/test_operation.py]: Added.

**Comment by dwierichs** [comment]
Thanks for the nice summary of the situation regarding the class attribute :100:

**Pull request merged**